launch/launch/actions/execute_local.py

```diff
@@ -464,9 +480,35 @@ def printer(context, msg, timeout_substitutions):
             ],
             cancel_on_shutdown=False,
         )
+        def kill_subprocesses(
+            context,
+            timeout_substitutions,
+            children=psutil.Process(
+                self._subprocess_transport.get_pid()).children(recursive=True)
+        ):
+            process_name = context.locals.process_name
+            for p in children:
+                try:
+                    p.send_signal(signal.SIGKILL)
+                except psutil.NoSuchProcess:
+                    continue
+                self.__logger.warn(
+                    f'subprocess[pid={p.pid}] of process[{process_name}] was not terminated after '
+                    f"'{perform_substitutions(context, timeout_substitutions)}' seconds of parent "
+                    f"being killed. "
+                    'Sending SIGKILL to subprocess directly.'
```

Comment:
Suggested rewording:

```suggestion
                    f'subprocess[pid={p.pid}] of process[{process_name}] did not terminate after '
                    f"'{perform_substitutions(context, timeout_substitutions)}' seconds since the parent "
                    f"was killed. "
                    'Sending SIGKILL to subprocess directly.'
```